ffi/diplomat/src/provider.rs

```diff
@@ -130,13 +130,13 @@ pub mod ffi {
         #[diplomat::rust_link(icu_provider_blob::BlobDataProvider, Struct)]
         #[allow(unused_variables)] // conditional on features
         pub fn create_from_byte_slice(
-            blob: &[u8],
+            blob: &'static [u8],
         ) -> DiplomatResult<Box<ICU4XDataProvider>, ICU4XError> {
```

Comment:
As discussed: we want a non-leaking version of this, but it is planned to be one that takes a `Box<[u8]>`.